src/ExadelMentorship.BusinessLogic/Services/MessageBus.cs

```diff
@@ -0,0 +1,63 @@
+﻿using ExadelMentorship.BusinessLogic.Interfaces.MessageBus;
+using ExadelMentorship.BusinessLogic.Models;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ExadelMentorship.BusinessLogic.Services
+{
+    public class MessageBus : IMessageProducer, IMessageConsumer
+    {
+        private RabbitMQSettings _rabbitMQSettings;
+        public MessageBus(IOptions<RabbitMQSettings> rabbitMQSettings)
+        {
+            _rabbitMQSettings = rabbitMQSettings.Value;
+        }
+
+        public void ReceiveMessage()
```

Comment:
Don't we return the message? Think of how to return the message and also, since this is an IO-bound operation it should be an async.

Reply:
can't understand how to take out 'message' from 'consumer'
<img width="581" alt="image" src="https://user-images.githubusercontent.com/59729680/174295681-7a2e7001-dab4-402c-9d06-041ad2fc5951.png">

Reply:
Glad you asked. Before TAP, .NET developer used callbacks for async programming. The formal name is [EAP](https://docs.microsoft.com/en-us/dotnet/standard/asynchronous-programming-patterns/event-based-asynchronous-pattern-eap). You can google how to convert event/callback code Task-based async code. You might want to use `TaskCompletionSource` (not sure though)

Reply:
added Action

Reply:
please check